### PR TITLE
Image refresh for arch

### PR DESCRIPTION
**Pull request by cockpituous**
Image refresh for arch
 * [x] image-refresh arch

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20230222-225736

**Comment by cockpituous** [comment]
Completed. Log: Error accessing /repos/cockpit-project/bots/pulls
  Status: 422
  Reason: Unprocessable Entity
  Response: {"message":"The specified issue is already attached to a pull request.","errors":[{"value":4446,"resource":"PullRequest","field":"issue","code":"invalid"}],"documentation_url":"https://docs.github.com/rest/reference/pulls#create-a-pull-request"}

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/arch-20230222-225656.log

**Comment by jelly** [comment]
TestConnection-testSocket

```
  curl (7.87.0-3 -> 7.88.1-3)
```

TestShutdownRestart.testBasic - it takes longer then expected to show the `New host` dialog

**Comment by jelly** [comment]
```
access("/etc/ssh/ssh_known_hosts", R_OK) = -1 ENOENT (No such file or directory)
pipe2([4, 5], 0)                        = 0
pipe2([6, 7], 0)                        = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13318, tv_nsec=828700782}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13318, tv_nsec=828729203}) = 0
futex(0x7feb66748320, FUTEX_WAKE_PRIVATE, 2147483647) = 0
getpid()                                = 1505
gettid()                                = 1505
pipe2([8, 9], 0)                        = 0
clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7feb65ec5ed0) = 1506
close(9)                                = 0
read(8, "", 8)                          = 0
close(8)                                = 0
close(5)                                = 0
close(7)                                = 0
poll([{fd=4, events=POLLIN|POLLERR|POLLHUP}, {fd=6, events=POLLIN|POLLERR|POLLHUP}], 2, -1


) = 1 ([{fd=6, revents=POLLHUP}])
read(6, "", 4096)                       = 0
close(6)                                = 0
poll([{fd=4, events=POLLIN|POLLERR|POLLHUP}, {fd=-1}], 2, -1) = 1 ([{fd=4, revents=POLLHUP}])
read(4, "", 4096)                       = 0
close(4)                                = 0
wait4(1506, [{WIFEXITED(s) && WEXITSTATUS(s) == 0}], 0, NULL) = 1506
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=1506, si_uid=0, si_status=0, si_utime=0, si_stime=0} ---
getpid()                                = 1505
write(1, "82\n", 382
)                     = 3
write(1, "\n{\"command\":\"authorize\",\"challen"..., 82
{"command":"authorize","challenge":"x-host-key","cookie":"session15051677515043"}) = 82
read(0, "\n", 3)                        = 1
read(0, "\n", 2)                        = 1
read(0, "\n", 1)                        = 1
ioctl(2, TCGETS, {c_iflag=ICRNL|IXON|IUTF8, c_oflag=NL0|CR0|TAB0|BS0|VT0|FF0|OPOST|ONLCR, c_cflag=B38400|CS8|CREAD, c_lflag=ISIG|ICANON|ECHO|ECHOE|ECHOK|IEXTEN|ECHOCTL|ECHOKE, ...}) = 0
gettimeofday({tv_sec=1677515043, tv_usec=760135}, NULL) = 0
newfstatat(AT_FDCWD, "/etc/localtime", {st_mode=S_IFREG|0644, st_size=114, ...}, 0) = 0
write(2, "cockpit-ssh-\33[1;32mMessage\33[0m: "..., 98cockpit-ssh-Message: 16:24:03.760: couldn't read control message: Bad message
) = 98
ioctl(2, TCGETS, {c_iflag=ICRNL|IXON|IUTF8, c_oflag=NL0|CR0|TAB0|BS0|VT0|FF0|OPOST|ONLCR, c_cflag=B38400|CS8|CREAD, c_lflag=ISIG|ICANON|ECHO|ECHOE|ECHOK|IEXTEN|ECHOCTL|ECHOKE, ...}) = 0
gettimeofday({tv_sec=1677515043, tv_usec=761603}, NULL) = 0
newfstatat(AT_FDCWD, "/etc/localtime", {st_mode=S_IFREG|0644, st_size=114, ...}, 0) = 0
write(2, "cockpit-ssh-\33[1;32mMessage\33[0m: "..., 128cockpit-ssh-Message: 16:24:03.761: cockpit-ssh 10.111.113.1: refusing to connect to unknown host: 10.111.113) = 128
write(2, ".1:22\n", 6.1:22
)                  = 6
write(1, "92\n", 392
)                     = 3
write(1, "\n{\"command\":\"init\",\"problem\":\"un"..., 92
{"command":"init","problem":"unknown-host","error":"unknown-host","auth-method-results":{}}) = 92
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13326, tv_nsec=540500747}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13326, tv_nsec=540893227}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13326, tv_nsec=541159332}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13326, tv_nsec=541255067}) = 0
poll([{fd=3, events=POLLIN}], 1, 0)     = 0 (Timeout)
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13326, tv_nsec=541750399}) = 0
write(3, "\1\0\0\0\0\0\0\0", 8)         = 8
futex(0x5653a8698010, FUTEX_WAKE_PRIVATE, 2147483647) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13326, tv_nsec=542491260}) = 0
clock_gettime(CLOCK_MONOTONIC, {tv_sec=13326, tv_nsec=542799480}) = 0
exit_group(2)                           = ?
+++ exited with 2 +++
[root@machine2 ~]#
```

**Comment by jelly** [comment]
Reported it https://github.com/curl/curl/issues/10633

**Comment by martinpitt** [comment]
I had a look at `TestHostSwitching.testBasic`. `ssh(1)` is fast:
```
# ssh admin@10.111.113.2
The authenticity of host '10.111.113.2 (10.111.113.2)' can't be established.
ED25519 key fingerprint is SHA256:saurX/J8uTqXdbKXd2rp3B9lxy8Gv+g1wSAsE4t4mx0.
This key is not known by any other names.
Are you sure you want to continue connecting (yes/no/[fingerprint])? ^C
```
This appears immediately. But with cockpit-ssh:
```
# yes '' | G_MESSAGES_DEBUG=all /usr/lib/cockpit/cockpit-ssh admin@10.111.113.2
72

{"command":"authorize","challenge":"*","cookie":"session6621677572411"}cockpit-ssh-Message: 08:20:11.808: couldn't read control message: Bad message
(cockpit-ssh:662): cockpit-ssh-DEBUG: 08:20:11.808: cockpit-ssh admin@10.111.113.2: host argument 'admin@10.111.113.2', host '10.111.113.2', username 'admin', port '0'
[2023/02/28 08:20:11.809202, 3] ssh_config_parse_file:  Reading configuration data from /etc/ssh/ssh_config
[2023/02/28 08:20:11.809344, 1] ssh_session_has_known_hosts_entry:  Cannot access file /root/.ssh/known_hosts
[2023/02/28 08:20:11.809440, 1] ssh_session_has_known_hosts_entry:  Cannot access file /etc/ssh/ssh_known_hosts
[2023/02/28 08:20:11.809564, 1] ssh_session_has_known_hosts_entry:  Cannot find a known_hosts file
[2023/02/28 08:20:11.809710, 1] ssh_session_has_known_hosts_entry:  Cannot access file /root/.ssh/known_hosts
[2023/02/28 08:20:11.809850, 1] ssh_session_has_known_hosts_entry:  Cannot access file /etc/ssh/ssh_known_hosts
[2023/02/28 08:20:11.809997, 1] ssh_session_has_known_hosts_entry:  Cannot find a known_hosts file
(cockpit-ssh:662): cockpit-ssh-DEBUG: 08:20:11.810: cockpit-ssh admin@10.111.113.2: host not known in any local file, asking sssd

[ ... 8s pause ... ]

(cockpit-ssh:662): cockpit-ssh-DEBUG: 08:20:19.545: cockpit-ssh admin@10.111.113.2: sss_ssh_knownhostsproxy failed: exit code 0, output '', error ''
81

{"command":"authorize","challenge":"x-host-key","cookie":"session6621677572419"}cockpit-ssh-Message: 08:20:19.546: couldn't read control message: Bad message
(cockpit-ssh:662): cockpit-ssh-DEBUG: 08:20:19.546: cockpit-ssh admin@10.111.113.2: using known hosts file (null); host known: 0; connect to unknown hosts: 0
cockpit-ssh-Message: 08:20:19.547: cockpit-ssh admin@10.111.113.2: refusing to connect to unknown host: 10.111.113.2:22
92

{"command":"init","problem":"unknown-host","error":"unknown-host","auth-method-results":{}}
```

There is a long pause for asking sssd. There are no sssd units running, e.g. `sssd-ifp.service` is disabled/stopped.

On the current main image:
```
# time sss_ssh_knownhostsproxy --pubkey --port 22 10.111.113.2
real	0m1.215s
```
but on this refresh:
```
# time sss_ssh_knownhostsproxy --pubkey --port 22 10.111.113.2

real	0m7.755s
user	0m0.000s
sys	0m0.010s
``` 

Confirmation: With this diff, the test passes, and the dialog appears quickly:
```diff
--- test/verify/check-shell-host-switching
+++ test/verify/check-shell-host-switching
@@ -140,6 +140,8 @@ class TestHostSwitching(MachineCase, HostSwitcherHelpers):
         m2.execute("hostnamectl set-hostname machine2")
         m3.execute("hostnamectl set-hostname machine3")
 
+        m1.execute("chmod a-x /usr/bin/sss_ssh_knownhostsproxy")
+
         # This should all work without being admin on machine1
         self.login_and_go(superuser=False)
```

As sssd did not change in this image, I have a feeling that this is an incompatibility with/regression from systemd 253, as there are no sssd units running at all. (They can't be started easily, as there is no configuration).

However, even after `systemctl log-level debug` there are no journal messages when running the command. But `strace -fvvtts1024` gives a further hint:

```
08:39:28.985507 connect(3, {sa_family=AF_UNIX, sun_path="/run/systemd/resolve/io.systemd.Resolve"}, 42) = 0
08:39:28.985716 brk(0x55b968717000)     = 0x55b968717000
08:39:28.985831 sendto(3, "{\"method\":\"io.systemd.Resolve.ResolveAddress\",\"parameters\":{\"address\":[10,111,113,2],\"family\":2,\"flags\":0}}\0", 108, MSG_DONTWAIT|MSG_NOSIGNAL, NULL, 0) = 108
08:39:28.986153 recvfrom(3, 0x55b9686f66b0, 131080, MSG_DONTWAIT, NULL, NULL) = -1 EAGAIN (Resource temporarily unavailable)
08:39:28.986304 ppoll([{fd=3, events=POLLIN}], 1, {tv_sec=119, tv_nsec=999827000}, NULL, 8) = 1 ([{fd=3, revents=POLLIN}], left {tv_sec=112, tv_nsec=459600604})
08:39:36.526818 recvfrom(3, "{\"error\":\"io.systemd.Resolve.MaxAttemptsReached\",\"parameters\":{}}\0", 131080, MSG_DONTWAIT, NULL, NULL) = 66
```

dns

**Comment by jelly** [comment]
Opened a PR here https://github.com/cockpit-project/cockpit/pull/18425 and triggered the bots.

**Comment by jelly** [comment]
Reported upstream as https://github.com/systemd/systemd/issues/26620

**Comment by jelly** [comment]
Ok, let's re-refresh, disable mDNS now for and the tests should go green :+1:

**Comment by cockpituous** [comment]
image-refresh arch done: https://github.com/cockpit-project/bots/commits/image-refresh-arch-20230301-085537

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/arch-20230301-085555.log

File: images/scripts/arch.setup [hunk 1]
```diff
@@ -99,6 +99,10 @@ no-auto-default=*
 unmanaged-devices=interface-name:eth0
 EOL
 
+# Disable mDNS for systemd-resolved
```
Comment (mvollmer) [inline images/scripts/arch.setup:102]: I think we should explain here why we switch it off (and in the commit message why it is needed now and not earlier).

Comment (jelly) [reply images/scripts/arch.setup]: Added some explanation to the comment.

**Review by mvollmer — APPROVED** [review]

**Pull request merged**